### PR TITLE
[bugfix] TPU + all_gather + SingleTPU shouldn't call xm.all_gather

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #<issue_number>

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/tpu.py [hunk 1]
```diff
@@ -46,4 +46,7 @@ def all_gather(self, tensor: torch.Tensor, group: Optional[Any] = None, sync_gra
         Return:
             A tensor of shape (world_size, batch, ...)
         """
-        return xm.all_gather(tensor, group=group, sync_grads=sync_grads)
+        # todo: Add support for backward with all_gather
+        if torch.distributed.is_initialized():
+            return xm.all_gather(tensor, group=group, sync_grads=sync_grads)
+        return tensor
```
Comment (SeanNaren) [inline pytorch_lightning/accelerators/tpu.py:52]: Code cov complaining about no test, probably want to add one to make sure this case is covered in single TPU!

File: CHANGELOG.md [hunk 2]
```diff
@@ -86,6 +86,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed duplicate logs appearing in console when using the python logging module ([#5509](https://github.com/PyTorchLightning/pytorch-lightning/pull/5509), [#6275](https://github.com/PyTorchLightning/pytorch-lightning/pull/6275))
 
 
+- Fixed `all_gather` on SingleTPU shouldn't call `all_gather` ([#6296](https://github.com/PyTorchLightning/pytorch-lightning/pull/6296))
```
Comment (carmocca) [inline CHANGELOG.md:89]:
```suggestion
- Fixed `SingleTPU` calling `all_gather` ([#6296](https://github.com/PyTorchLightning/pytorch-lightning/pull/6296))
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
can we add a test for it?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/accelerators/tpu.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/tpu.py:52]: The test cover this part but coverage is currently not supported for TPU.

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**